mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -636,4 +638,137 @@ private ArtifactRepository getArtifactRepository(String runId) {
+  }
+
+  /**
+   * Return the latest model version for each stage..
+   * The current stages are: [None, Staging, Production, Archived].
+   *
+   *    <pre>
+   *        ModelVersionDetailed details = getLatestVersions("model", "Staging");
+   *
+   *        System.out.println("Model Name: " + details.getModelVersion()
+   *                                                   .getRegisteredModel()
+   *                                                   .getName());
+   *        System.out.println("Model Version: " + details.getModelVersion().getVersion());
+   *        System.out.println("Current Stage: " + details.getCurrentStage());
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param stage The name of the stage
+   * @return The latest model version
+   *         {@link org.mlflow.api.proto.ModelRegistry.ModelVersionDetailed}
+   */
+  public ModelVersionDetailed getLatestVersions(@Nonnull String modelName, @Nonnull String stage) {
+    String json = sendPost("registered-models/get-latest-versions",
+            mapper.makeGetLatestVersion(modelName, stage));
+
+    GetLatestVersions.Response response =  mapper.toGetLatestVersionsResponse(json);
+
+    if (response.getModelVersionsDetailedCount() < 1) {
+      throw new MlflowClientException(String.format("The model %s does not exist", modelName));
+    }
+
+    if (response.getModelVersionsDetailedCount() > 1) {
+      throw new MlflowClientException(String.format("More than one one model version exists" +
+              "for the model %s for the stage %s", modelName, stage));
+    }
+    return response.getModelVersionsDetailedList().get(0);
+  }
+
+  /**
+   * Return the model URI containing for the given model version. The model URI can be used
+   * to download the model version artifacts.
+   *
+   * For example, the model URI
+   *
+   *    <pre>
+   *        String modelUri = getModelVersionDownloadUri("model", 0);
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param version The version number of the model
+   * @return The specified model version's URI.
+   */
+  public String getModelVersionDownloadUri(@Nonnull String modelName, long version) {
+    String json = sendPost("model-versions/get-download-uri",
+            mapper.makeGetModelVersionDownloadUri(modelName, version));
+    return mapper.toGetModelVersionDownloadUriResponse(json);
+  }
+
+  /**
+   * Return a local file or directory containing all artifacts within the given registered model
+   * version. The method will download the model version artifacts to the local file system.
+   *
+   *    <pre>
+   *        File modelVersionFile = downloadModelVersion("model", 0);
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param version The version number of the model
+   * @return A the local file or directory ({@ java.io.File}) containing model artifacts
+   */
+  public File downloadModelVersion(@Nonnull String modelName, long version) {
```

Comment:
were you able to test this against the databricks model registry (which uses dbfs paths for the model download uri)?

Reply:
Good point. Not yet. This was only tested against the local mlflow server so far. We should add an integration test that can ensure the DBFS implementation works.

Reply:
I have manually test this against the dogfood stage right now. I will create a follow-up task to implement an integration test in the databricks universe